### ISSUE TITLE
[Scaffold] Scan QR Screen

### ISSUE THREAD

**Issue by pacu** [zcash/secant-android-wallet#312]
Rough implementation of Scanning feature.

- Permission prompt, with fallback to open settings if the user chose "don't show this again"
- camera is full screen (we probably end up using the jetpack camerax library for this)
- user can turn torch on
- back button
- Connect with zxing to parse the QR code
- Potentially in-scope: Verifying that a naked Zcash address is valid using the Synchronizer's API to validate an address 
- Error text is shown when a QR is recognized but the contained address is not valid
- Out of scope: parsing a Zip-321 URI (blocked by https://github.com/zcash/zcash-android-wallet-sdk/issues/397)
